src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -596,19 +596,19 @@ int32_t AppleCryptoNative_SslSetEnabledCipherSuites(SSLContextRef sslContext, co
     // Max numCipherSuites is 2^16 (all possible cipher suites)
     assert(numCipherSuites < (1 << 16));
 
-#if !defined(TARGET_MACCATALYST) && !defined(TARGET_IOS) && !defined(TARGET_TVOS)
+#if !defined(TARGET_IOS) && !defined(TARGET_TVOS)
```

Comment:
```suggestion
#if !defined(TARGET_CPU_ARM64) && !defined(TARGET_IOS) && !defined(TARGET_TVOS)
```

So, Apple headers are still broken and return `sizeof(SSLCipherSuite) == sizeof(uint32_t)` for MacCatalyst arm64. The simplest fix is to use the `else` (`uint16_t`) branch on any arm64 architecture.

Here's the additional rationale why the headers are broken: The system libraries that the app links to at runtime are identical for macOS and Mac Catalyst. Some of the APIs do runtime checks to adjust behavior but that's not the case for `SSLSetEnabledCiphers`. On macOS the API provably takes `uint16_t` and that's what is specified correctly in the system headers. For Mac Catalyst arm64 the `SSLSetEnabledCiphers` was already long obsolete by the time the arm64 architecture was added so Apple probably didn't bother with changing the type in the header.

Reply:
Sounds good though `TARGET_CPU_ARM64 ` is coming from `TargetConditionals.h` and I'm not sure it's available here. We already have our own define `TARGET_ARM64` that we're already using in other parts of the PAL so better to use that one.

Reply:
`TargetConditionals.h` is included but I am fine with either one as long as it works. I knew there was some other `#define` but I was not able to find where it is defined :-)